crates/primitives/starknet/src/transaction/types.rs

```diff
@@ -236,19 +229,28 @@ pub struct DeployAccountTransaction {
     pub max_fee: Felt252Wrapper,
 }
 
-impl DeployAccountTransaction {
-    /// converts the transaction to a [Transaction] object
-    pub fn from_deploy(self, chain_id: Felt252Wrapper) -> Result<Transaction, TransactionConversionError> {
-        let salt_as_felt: StarkFelt = StarkFelt(self.salt.into());
-        let stark_felt_vec: Vec<StarkFelt> = self.calldata.clone()
+impl TryFrom<(DeployAccountTransaction, Felt252Wrapper)> for Transaction {
```

Comment:
how idiomatic is that kind of pattern ? I find it quite misleading and not sure that rustdoc would make it easy to understand